ivy_models/convnext/convnext.py

```diff
@@ -53,30 +93,35 @@ def _build(self, *args, **kwargs):
             self.downsample_layers.append(downsample_layer)
 
         self.stages = []
-        dp_rates = [x for x in ivy.linspace(0, self.drop_path_rate, sum(self.depths))]
+        dp_rates = [x for x in ivy.linspace(
+            0, self.spec.drop_path_rate, sum(self.spec.depths))]
         cur = 0
         for i in range(4):
             stage = ivy.Sequential(
                 *[
                     ConvNeXtBlock(
-                        dim=self.dims[i],
+                        dim=self.spec.dims[i],
                         drop_path=dp_rates[cur + j],
-                        layer_scale_init_value=self.layer_scale_init_value,
+                        layer_scale_init_value=self.spec.layer_scale_init_value,
                     )
-                    if self.version == 1
+                    if self.spec.version == 1
                     else ConvNeXtV2Block(
-                        dim=self.dims[i],
+                        dim=self.spec.dims[i],
                         drop_path=dp_rates[cur + j],
-                        layer_scale_init_value=self.layer_scale_init_value,
+                        layer_scale_init_value=self.spec.layer_scale_init_value,
                     )
-                    for j in range(self.depths[i])
+                    for j in range(self.spec.depths[i])
                 ]
             )
             self.stages.append(stage)
-            cur += self.depths[i]
+            cur += self.spec.depths[i]
 
-        self.norm = ivy.LayerNorm([self.dims[-1]], eps=1e-6)
-        self.head = ivy.Linear(self.dims[-1], self.num_classes)
+        self.norm = ivy.LayerNorm([self.spec.dims[-1]], eps=1e-6)
+        self.head = ivy.Linear(self.spec.dims[-1], self.spec.num_classes)
+
+    @classmethod
+    def get_spec_class(self):
+        return ConvNeXtSpec
 
     def _forward(self, x):
         for i in range(4):
```

Comment:
side note: not sure if you're working on adding `data_format` dual support here, but if you are, the forward pass needs an additional check (ref `resnet` if I remember correctly) thanks!